lib/src/clojure_lsp/feature/file_management.clj

```diff
@@ -84,37 +85,56 @@
-                                   ;; TODO: switch from nested-loop join to hash join
-                                   (keep #(q/find-definition project-analysis % db))
-                                   (filter (fn [d]
-                                             ;; TODO: this excludes "private calls", but they are counted in code lens, so maybe shouldn't exclude
-                                             (and (not (:private d))
-                                                  ;; TODO: is this extra work? The definition came from project-analysis, so should be on the source-paths.
-                                                  (some #(string/starts-with? (:filename d) %) source-paths))))
-                                   (map :filename))]
-    (disj (set (concat references-filenames definitions-filenames))
-          filename)))
+        project-analysis (into {}
+                               (q/filter-project-analysis-xf db)
+                               (dissoc (:analysis @db) filename)) ;; don't notify self
+        incoming-filenames (when (seq changed-var-definitions)
+                             (let [def-signs (->> changed-var-definitions
+                                                  (map q/var-definition-signatures)
+                                                  (apply set/union))]
+                               (into #{}
+                                     (comp
+                                       (mapcat val)
+                                       (filter #(identical? :var-usages (:bucket %)))
+                                       (filter #(contains? def-signs (q/var-usage-signature %)))
+                                       (map :filename))
+                                     project-analysis)))
+        outgoing-filenames (when (seq changed-var-usages)
+                             (let [usage-signs->langs (->> changed-var-usages
+                                                              ;; TODO: do we really care if lang doesn't match?
+                                                           (reduce (fn [result usage]
+                                                                     (assoc result (q/var-usage-signature usage) (q/elem-langs usage)))
+                                                                   {}))]
+                               (into #{}
+                                     (comp
+                                       (mapcat val)
+                                       (filter #(identical? :var-definitions (:bucket %)))
+                                       (filter #(when-let [usage-langs (some usage-signs->langs (q/var-definition-signatures %))]
+                                                  (some usage-langs (q/elem-langs %))))
+                                          ;; TODO: this excludes "private calls", but they are counted in code lens, so maybe shouldn't exclude
```

Comment:
I think the purpose of this at the time, was to exclude private calls since they are from the same file and we don't want to lint the same file, only public calls, does that makes sense for you?

Reply:
you are right, we should lint private calls even if from other files like test ones which is pretty common, we even have on clojure-lsp project some of them

Reply:
That's a good point... even more reason to lint different-file private calls.